diff --git a/flasher_stub/Makefile b/flasher_stub/Makefile
index 4e4f4d9..c794e29 100644
--- a/flasher_stub/Makefile
+++ b/flasher_stub/Makefile
@@ -71,7 +71,2 @@ STUB_ELF_32H2 = $(BUILD_DIR)/$(STUB)_32h2.elf

-STUBS_ELF =
-ifneq ($(WITHOUT_ESP8266),1)
-STUBS_ELF += \
-  $(STUB_ELF_8266)
-endif

@@ -124,15 +124,12 @@ LDLIBS = -lgcc

-$(STUB_ELF_8266): $(SRCS) $(SRCS_8266) $(BUILD_DIR) ld/stub_8266.ld | Makefile
-	@echo "  CC(8266)   $^ -> $@"
-	$(Q) $(CROSS_8266)gcc $(CFLAGS) -DESP8266=1 -Tstub_8266.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)

 $(STUB_ELF_32): $(SRCS) $(BUILD_DIR) ld/stub_32.ld | Makefile
 	@echo "  CC(32)   $^ -> $@"
-	$(Q) $(CROSS_32)gcc $(CFLAGS) -DESP32=1 -Tstub_32.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)
+	cp ../../target/xtensa-esp32-none-elf/release/flasher-stub build/stub_flasher_32.elf
 
 $(STUB_ELF_32S2): $(SRCS) $(BUILD_DIR) ld/stub_32s2.ld
 	@echo "  CC(32S2)   $^ -> $@"
-	$(Q) $(CROSS_32S2)gcc $(CFLAGS) -DESP32S2=1 -Tstub_32s2.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)
+	cp ../../target/xtensa-esp32s2-none-elf/release/flasher-stub build/stub_flasher_32s2.elf
 
 $(STUB_ELF_32S3_BETA_2): $(SRCS) $(BUILD_DIR) ld/stub_32s3_beta_2.ld
 	@echo "  CC(32S3)   $^ -> $@"
@@ -138,11 +135,11 @@ $(STUB_ELF_32S3_BETA_2): $(SRCS) $(BUILD_DIR) ld/stub_32s3_beta_2.ld

 $(STUB_ELF_32S3): $(SRCS) $(BUILD_DIR) ld/stub_32s3.ld
 	@echo "  CC(32S3)   $^ -> $@"
-	$(Q) $(CROSS_32S3)gcc $(CFLAGS) -DESP32S3=1 -Tstub_32s3.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)
+	cp ../../target/xtensa-esp32s3-none-elf/release/flasher-stub build/stub_flasher_32s3.elf
 
 $(STUB_ELF_32C3): $(SRCS) $(BUILD_DIR) ld/stub_32c3.ld
 	@echo "  CC(32C3)   $^ -> $@"
-	$(Q) $(CROSS_ESPRISCV32)gcc $(CFLAGS_ESPRISCV32) -DESP32C3=1 -Tstub_32c3.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)
+	cp ../../target/riscv32imc-unknown-none-elf/release/flasher-stub build/stub_flasher_32c3.elf
 
 $(STUB_ELF_32C6BETA): $(SRCS) $(BUILD_DIR) ld/stub_32c6_beta.ld
 	@echo "  CC(32C6BETA)   $^ -> $@"
@@ -158,9 +155,9 @@ $(STUB_ELF_32H2_BETA_2): $(SRCS) $(BUILD_DIR) ld/stub_32h2_beta_2.ld

 $(STUB_ELF_32C2): $(SRCS) $(BUILD_DIR) ld/stub_32c2.ld
 	@echo "  CC(32C2)   $^ -> $@"
-	$(Q) $(CROSS_ESPRISCV32)gcc $(CFLAGS_ESPRISCV32) -DESP32C2=1 -Tstub_32c2.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)
+	cp ../../target/riscv32imc-unknown-none-elf/release/flasher-stub build/stub_flasher_32c2.elf
 
 $(STUB_ELF_32C6): $(SRCS) $(BUILD_DIR) ld/stub_32c6.ld
 	@echo "  CC(32C6)   $^ -> $@"
-	$(Q) $(CROSS_ESPRISCV32)gcc $(CFLAGS_ESPRISCV32) -DESP32C6=1 -Tstub_32c6.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)
+	cp ../../target/riscv32imac-unknown-none-elf/release/flasher-stub build/stub_flasher_32c6.elf

@@ -170,4 +167,4 @@ $(STUB_ELF_32H2): $(SRCS) $(BUILD_DIR) ld/stub_32h2.ld
 	@echo "  CC(32H2)   $^ -> $@"
-	$(Q) $(CROSS_ESPRISCV32)gcc $(CFLAGS_ESPRISCV32) -DESP32H2=1 -Tstub_32h2.ld -Wl,-Map=$(@:.elf=.map) -o $@ $(filter %.c, $^) $(LDLIBS)
+	cp ../../target/riscv32imac-unknown-none-elf/release/flasher-stub build/stub_flasher_32h2.elf

 clean:
	$(Q) rm -rf $(BUILD_DIR)
